src/toxcore/messenger/packet/group/mod.rs

```diff
@@ -0,0 +1,80 @@
+/*! The implementation of group chat packets.
+*/
+
+mod invite;
+
+pub use self::invite::*;
+
+use nom::be_u8;
+use crate::toxcore::binary_io::*;
+use crate::toxcore::crypto_core::*;
+
+/// Length of group chat unique bytes
+pub const GROUP_UID_BYTES: usize = 32;
+
+/// Unique id used in group chat
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct GroupUID([u8; GROUP_UID_BYTES]);
+
+impl GroupUID {
+    /// Create new object
+    pub fn new() -> Self {
+        let mut array = [0; GROUP_UID_BYTES];
+        randombytes_into(&mut array);
+        GroupUID(array)
+    }
+
+    /// Custom from_slice function of GroupUID
+    pub fn from_slice(bs: &[u8]) -> Option<GroupUID> {
+        if bs.len() != GROUP_UID_BYTES {
+            return None
+        }
+        let mut n = GroupUID([0; GROUP_UID_BYTES]);
+        for (ni, &bsi) in n.0.iter_mut().zip(bs.iter()) {
+            *ni = bsi
+        }
+        Some(n)
+    }
+}
+
+impl FromBytes for GroupUID {
```

Comment:
Why there is no `ToBytes`?

Reply:
Yeah, why?

Reply:
`GroupUID` is not a packet type. It is not used as an independent type for serde, only used within packets of group chat.
So, I think there is no need to provide `ToBytes` to this type.
This is same to `PublicKey` type, it has only `FromBytes`.

Reply:
How can we send it then?

Reply:
https://github.com/tox-rs/tox/blob/20ec8d4ae846cd1bd0b9ca125890068a9b00ed36/src/toxcore/messenger/packet/group/invite.rs#L51